### PR TITLE
ci: use workflow templates

### PR DESCRIPTION
**Pull request by n8maninger**
No support for deb packaging, homebrew, or MySQL tests

Depends on https://github.com/SiaFoundation/workflows/pull/7

### CODE REVIEW

File: .github/workflows/test.yml
Comment (ChrisSchinnerl) [inline .github/workflows/test.yml:1]: Tbh not having mysql tests is a bit of a dealbreaker for me. I'd hold off on merging this until we have extended the common workflow with support for MySQL

File: .github/workflows/publish.yml
Comment (ChrisSchinnerl) [inline .github/workflows/publish.yml:1]: We should probably prioritize getting linux and homebrew builds running. It would be awkward to stop releasing on linux mid-beta.

File: .github/workflows/test.yml
Comment (n8maninger) [reply .github/workflows/test.yml:1]: Im not certain the common workflow should support MySQL. None of the other apps will ever support it.

Comment (n8maninger) [reply .github/workflows/test.yml:1]: We can probably append the MySQL job as an extra step at the end of the workflow file

Comment (ChrisSchinnerl) [reply .github/workflows/test.yml:1]: Yeah I wouldn't spin up a mysql database in the common workflow but it would have been cool if we could inject a generic setup step and custom environment variables into the common workflow.

Maybe something like this: https://github.com/aws-deadline/.github/pull/15
It seems to trigger a workflow iff it is defined which we could use as an optional setup step.

**Review by peterjan — COMMENTED** [review]
Nothing to add to Chris's comments, looks good to me 👍

File: .github/workflows/test.yml
Comment (ChrisSchinnerl) [inline .github/workflows/test.yml:14]: Blocker to update the branches here to `master` after merging https://github.com/SiaFoundation/workflows/pull/7

**Comment by ChrisSchinnerl** [comment]
@n8maninger reverted the changes to the test.yml file and tested the homebrew action by pasting it into the test action and running it once with a valid tag and once with an invalid one.

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**
